shared/src/question-generators/propositional-logic/minimize.ts

```diff
@@ -0,0 +1,189 @@
+import {
+  FreeTextFeedbackFunction,
+  FreeTextFormatFunction,
+  FreeTextQuestion,
+  QuestionGenerator,
+} from "@shared/api/QuestionGenerator"
+import { serializeGeneratorCall } from "@shared/api/QuestionRouter"
+import { minimizeExpressionDNF } from "@shared/question-generators/propositional-logic/minimizeDNF"
+import {
+  getTypingAids,
+  getTypingAidsVars,
+  variableNames,
+} from "@shared/question-generators/propositional-logic/utils"
+import { _ } from "@shared/utils/generics"
+import {
+  compareExpressions,
+  generateRandomExpression,
+  ParserError,
+  PropositionalLogicParser,
+  SyntaxTreeNodeType,
+  tokenToLatex,
+} from "@shared/utils/propositionalLogic"
+import Random from "@shared/utils/random"
+import { t, tFunctional, Translations } from "@shared/utils/translations"
+
+const translations: Translations = {
+  en: {
+    name: "Minimize boolean expression",
+    description: "Minimize the normal form of a boolean expression",
+    text: "Given the boolean expression \\[\\varPhi={{0}}\\] Let $\\varPhi^*$ be the minimized expression of $\\varPhi$ in **{{1}}**.",
```

Comment:
I see here different tasks: Minimizing an arbitrary boolean expression to another arbitrary expression and minimizing a normal form to a other normal form. The actual task seems to be: given an arbitrary expression, state a minimal normal form. The text should also state what to do. currently it only states what \varPhi^* is and not that the user should compute it.

However, if this is the task, it seems to be basically the same as the other task about normal forms, except that the solution should be minimal. This would make an excellent variant of the other task, where you randomly choose if the solution should be minimal, (or maybe have a parameter).

Reply:
I ll write the task for clear, but I would keep it as a separate generator because the question is (in my opinion) much more difficult then just computing a normal form of any length.

Reply:
ok, is fine for me.